### PR TITLE
d/jak2: First few files, `mood-tables` and `vol-h`

### PR DESCRIPTION
**Pull request by xTVaser**
Did a bunch of work in `mood` and `mood-funcs` but they aren't 100% done yet (let 2 failing functions).  So I didn't add those yet.

Did some housekeeping around running the memory dumper and the default config options as well.

### CODE REVIEW

File: goal_src/jak2/engine/geometry/vol-h.gc [hunk 1]
```diff
@@ -7,3 +7,144 @@
...
+  (format #t "~1Tvolume-type: ~A~%" (-> obj volume-type))
+  (format #t "~1Tpoint-count: ~D~%" (-> obj point-count))
+  (format #t "~1Tnormal-count: ~D~%" (-> obj normal-count))
+  (format #t "~1Tfirst-point: #<vector @ #x~X>~%" (-> obj first-point))
+  (format #t "~1Tfirst-normal: #<vector @ #x~X>~%" (-> obj first-normal))
+  (format #t "~1Tnum-planes: ~D~%" (-> obj num-planes))
+  (format #t "~1Tplane: #x~X~%" (-> obj plane))
+  (label cfg-4)
+  obj
+  )
```
Comment (water111) [inline goal_src/jak2/engine/geometry/vol-h.gc:45]: you can remove this inspect method

File: goal_src/jak2/engine/geometry/vol-h.gc [hunk 2]
```diff
@@ -7,3 +7,144 @@
...
+  (format #t "~1Tprocess: ~A~%" (-> obj process))
+  (format #t "~1Tpos-vol-count: ~D~%" (-> obj pos-vol-count))
+  (format #t "~1Tpos-vol[32] @ #x~X~%" (-> obj pos-vol))
+  (format #t "~1Tneg-vol-count: ~D~%" (-> obj neg-vol-count))
+  (format #t "~1Tneg-vol[32] @ #x~X~%" (-> obj neg-vol))
+  (format #t "~1Tdebug-point: ~A~%" (-> obj debug-point))
+  (format #t "~1Tdebug-normal: ~A~%" (-> obj debug-normal))
+  (label cfg-4)
+  obj
+  )
```
Comment (water111) [inline goal_src/jak2/engine/geometry/vol-h.gc:84]: same here

File: goal_src/jak2/engine/gfx/mood/mood-tables.gc [hunk 3]
```diff
@@ -7,3 +7,867 @@
...
+                                 )
+  )
+
+(define *no-cloud-mood-channel-group*
+  (new 'static 'mood-channel-group
+    :data (new 'static 'inline-array mood-channel 4
+      (new 'static 'mood-channel
+        :vecs (new 'static 'inline-array vector4 6
+          (new 'static 'vector4)
+          (new 'static 'vector4 :data (new 'static 'array float 4 0.0 0.2 0.4 1.0))
```
Comment (water111) [inline goal_src/jak2/engine/gfx/mood/mood-tables.gc:544]: Can you add `:score -1` to the `data` in `deftype vector4`? I think that will make these look a lot nicer, and I meant to do this originally.

File: goal_src/jak2/engine/gfx/mood/mood-tables.gc [hunk 4]
```diff
@@ -7,3 +7,867 @@
...
+  (set! (-> arg0 mood-channel-group) (new 'global 'mood-channel-group))
+  (set! (-> arg0 mood-clouds) (new 'global 'mood-clouds))
+  (set! (-> arg0 mood-direction-table) *mood-direction-table*)
+  (set! (-> arg0 mood-sky-table) *mood-sky-table*)
+  (set! (-> arg0 mood-interp-table) (the-as basic *mood-interp-table*))
+  (mem-copy! (the-as pointer (-> arg0 mood-fog-table)) (the-as pointer *no-cloud-clear-mood-fog-table*) 384)
+  (mem-copy! (the-as pointer (-> arg0 mood-color-table)) (the-as pointer *no-cloud-mood-color-table*) 256)
+  (mem-copy! (the-as pointer (-> arg0 mood-channel-group)) (the-as pointer *no-cloud-mood-channel-group*) 384)
+  (mem-copy! (the-as pointer (-> arg0 mood-clouds)) (the-as pointer *clouds-250*) 8)
+  (set! (-> arg0 fogs 0) (the-as mood-fog *no-cloud-haze-mood-fog-table*))
```
Comment (water111) [inline goal_src/jak2/engine/gfx/mood/mood-tables.gc:820]: This is my fault - `fogs` in `mood-control` should be `mood-fog-table`

File: goal_src/jak2/engine/gfx/mood/mood-tables.gc [hunk 5]
```diff
@@ -7,3 +7,867 @@
...
+  (set! (-> arg0 fogs 1) (the-as mood-fog *no-cloud-clear-mood-fog-table*))
+  (set! (-> arg0 fogs 2) (the-as mood-fog *no-cloud-smog-mood-fog-table*))
+  (set! (-> arg0 fogs 3) (the-as mood-fog *overcast-fog-mood-fog-table*))
+  (set! (-> arg0 fogs 4) (the-as mood-fog *overcast-clear-mood-fog-table*))
+  (set! (-> arg0 fogs 5) (the-as mood-fog *overcast-rain-mood-fog-table*))
+  (set! (-> arg0 fogs 6) (the-as mood-fog *storm-fog-mood-fog-table*))
+  (set! (-> arg0 fogs 7) (the-as mood-fog *storm-clear-mood-fog-table*))
+  (set! (-> arg0 fogs 8) (the-as mood-fog *storm-rain-mood-fog-table*))
+  (set! (-> arg0 colors 0) (the-as mood-color *no-cloud-mood-color-table*))
```
Comment (water111) [inline goal_src/jak2/engine/gfx/mood/mood-tables.gc:829]: `colors` should be `mood-color-table`

File: goal_src/jak2/engine/gfx/mood/mood-tables.gc [hunk 6]
```diff
@@ -7,3 +7,867 @@
...
+  (set! (-> arg0 colors 1) (the-as mood-color *overcast-mood-color-table*))
+  (set! (-> arg0 colors 2) (the-as mood-color *storm-mood-color-table*))
+  (set! (-> arg0 channels 0) (the-as mood-channel *no-cloud-mood-channel-group*))
```
Comment (water111) [inline goal_src/jak2/engine/gfx/mood/mood-tables.gc:832]: I was also wrong here

File: goal_src/jak2/engine/gfx/mood/mood-tables.gc [hunk 7]
```diff
@@ -7,3 +7,867 @@
...
+  (set! (-> arg0 channels 1) (the-as mood-channel *overcast-mood-channel-group*))
+  (set! (-> arg0 channels 2) (the-as mood-channel *stormy-mood-channel-group*))
+  (set! (-> arg0 clouds 0) (the-as mood-clouds *clouds-000*))
```
Comment (water111) [inline goal_src/jak2/engine/gfx/mood/mood-tables.gc:835]: here too

File: goal_src/jak2/engine/gfx/mood/mood-tables2.gc [hunk 8]
```diff
@@ -7,3 +7,116 @@
...
+
+(define *overide-table* (new 'static 'mood-table))
+
+(init-overide-table *overide-table*)
+
+;; ERROR: function has no type analysis. Cannot decompile.
+
+;; ERROR: function has no type analysis. Cannot decompile.
+
+;; ERROR: function has no type analysis. Cannot decompile.
```
Comment (water111) [inline goal_src/jak2/engine/gfx/mood/mood-tables2.gc:117]: are these the broken ones due to far labels/ash stuff?

**Review by water111 — COMMENTED** [review]
look good, just a few suggestions

File: goal_src/jak2/engine/gfx/mood/mood-tables2.gc (hunk 8 above)
Comment (xTVaser) [reply goal_src/jak2/engine/gfx/mood/mood-tables2.gc]: nah just missed them

**Pull request merged**